Fixed issue

zeljkoAmiq opened the pull request:
After @mickaelistria points on this thread: https://github.com/eclipse-platform/eclipse.platform.debug/pull/142 revision
On each mouse move we should cache the mouse offset.
Whenever the console's document changes, we should update the links at the latest mouse offset (where the mouse was before the document change)

zeljkoAmiq committed:
Fixed issue

iloveeclipse commented on org.eclipse.ui.console/src/org/eclipse/ui/console/TextConsoleViewer.java:
```diff
@@ -105,7 +107,7 @@ public void documentAboutToBeChanged(DocumentEvent event) {
 		@Override
 		public void documentChanged(DocumentEvent event) {
-			updateLinks(event.fOffset);
+			updateLinks(mouseOffset);
```
I don't think this is correct, if document is changed, previous mouse offset is most likely wrong, so the change offset as it was before was OK.

zeljkoAmiq replied:
The change offset as it was before was causing the bug.
This was causing the hyperlink which was being hovered before the document change
to be exited (unavailable until you move the mouse again).

Please see the conversations here:
https://github.com/eclipse-platform/eclipse.platform.debug/issues/141
and here:
https://github.com/eclipse-platform/eclipse.platform.debug/pull/142

The updateLinks method takes care of the linkEntered/linkExited logic.
This makes sense to be done only for the link which is being hovered (at the latest mouseOffset).

That's the only link which makes sense to be entered/exited after a document change.
Any other link will be entered/exited at mouseMove.

iloveeclipse reviewed (changes requested):
Beside the fact that the change itself seem to be not OK: If you have a fix, please provide **appropriate** commit message, not just "issue fixed".

zeljkoAmiq committed:
Merge branch 'master' into Console_hyperlinks_not_accessible_while_process_still_running_#142

iloveeclipse commented:
> Beside the fact that the change itself seem to be not OK: If you have a fix, please provide **appropriate** commit message, not just "issue fixed".

With the **commit** message I mean **commit**, not PR message.
But additionally you have now merged master to your branch and pushed that - please don't **merge** but rebase your branch on master, merge commits are not allowed.

In sum (assuming the change is correct) :
1) Please properly explain the change in the commit message. After that, please add the line `Fixes https://github.com/eclipse-platform/eclipse.platform/issues/529` to the commit message.
2) Rebase your commit on master instead of merging
3) Force push commit to your branch.

iloveeclipse commented on org.eclipse.ui.console/src/org/eclipse/ui/console/TextConsoleViewer.java:
```diff
@@ -93,6 +93,8 @@ public class TextConsoleViewer extends SourceViewer implements LineStyleListener
 
 	private boolean consoleAutoScrollLock = true;
 
+	private int mouseOffset;
```
Initial value should be -1, otherwise if mouse was never moved, changed code will still try to find and create a link at first line.

zeljkoAmiq committed:
Hyperlinks are not accessible while the process is still logging #141

When a document change event is triggered, updating the links in the console at the
document change offset results in exiting the link which was being hovered before
the console's document changed (while the running process is still logging into the console).

Updating the links (triggering the linkEntered/linkExited methods) should be done only for the
current mouse offset when a document changes because it's the only link which might be caused
to be entered / exited after a document change.

Fixes https://github.com/eclipse-platform/eclipse.platform.debug/issues/141

laeubi commented:
This repository was merged with https://github.com/eclipse-platform/eclipse.platform if this PR is still relevant please rebase it and push it to the new origin!
